prowler/providers/kubernetes/kubernetes_provider.py

```diff
@@ -237,7 +236,7 @@ def search_and_save_roles(
             logger.critical(
                 f"{error.__class__.__name__}[{error.__traceback__.tb_lineno}]: {error}"
             )
-            sys.exit(1)
+            raise error
```

Comment:
Add a TODO here to replace it with our custom exceptions once they are ready.

Reply:
Done

Reply:
Done